File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    var answer = true;

    // [실행] 버튼을 누르면 출력 값을 볼 수 있습니다.
    console.log('Hello Javascript')
    const arr  = s.split('')
    const stack = []
    for(const str of arr) {
        if(str === '(') {
            stack.push(str)
        }
        else {
            if(stack.length ===0 ) return false
            stack.pop();
        }
    }
    console.log(stack)
    if(stack.length === 0) {
        return true}
    else {
        return false
    }
}

File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.py

def solution(s):
    answer = True
    arr = []
    for x in list(s):
        if x == '(':
            arr.append(x)
        if x == ')':
            if len(arr) <= 0:
                return False
            arr.pop()
    if len(arr) == 0:
        return True
    else:
        return False
    

    return True